src/bblocks_data_importers/who/ghed.py

```diff
@@ -0,0 +1,190 @@
+import io
+import os
+import numpy as np
+from requests.exceptions import RequestException
+
+from bblocks_data_importers.config import logger, DataFormatError
+from bblocks_data_importers.protocols import DataImporter
+
+URL: str = "https://apps.who.int/nha/database/Home/IndicatorsDownload/en"
+
+
+class GHED(DataImporter):
+    """Importer for the GHED database from WHO.
+
+    The Global Health Expenditure Database (GHED) provides comparable data on health expenditure
+    across various countries and years. See more details and access the raw data at: https://apps.who.int/nha/database
+
+    This importer provides functionality to easily get data and metadata from the GHED database.
+
+    Usage:
+
+    First instantiate an importer object:
+    >>> ghed = GHED()
+
+    get the data and metadata:
+    >>> data = ghed.get_data()
+    >>> metadata = ghed.get_metadata()
+
+    Download the raw data to disk:
+    >>> ghed.download_data(path = "some_path")
+
+
+    This object caches the data as objects attributes to avoid multiple requests to the database.
+    You can clear the cached data using the `clear_cache method`
+    >>> ghed.clear_cache()
+    """
+
+    def __init__(self):
+        self._raw_data: io.BytesIO | None = None
+        self._data: pd.DataFrame | None = None
+        self._metadata: pd.DataFrame | None = None
+
+        # TODO: Add functionality to use a local file instead of downloading the data
+
+    @staticmethod
+    def _extract_raw_data() -> io.BytesIO:
+        """Extract the raw data from the GHED database"""
+
+        logger.info("Extracting data from GHED database")
+
+        try:
+            response = requests.get(URL)
+            response.raise_for_status()  # Raises an error for HTTP codes 4xx/5xx
+            return io.BytesIO(response.content)
+
+        except RequestException as e:
+            raise ConnectionError(f"Error extracting data: {e}")
+
+    @staticmethod
+    def _format_raw_data(raw_data: io.BytesIO) -> pd.DataFrame:
+        """Format the raw data from the GHED database
+
+        Args:
+            raw_data: Raw data extracted from the GHED database. Use the `_extract_raw_data` method to get this data
+        """
+
+        try:
+            data_df = (
+                pd.read_excel(raw_data, sheet_name="Data")
+                .drop(columns=["region", "income"])
+                .melt(id_vars=["country", "code", "year"], var_name="indicator_code")
+                .rename(columns={"country": "country_name", "code": "iso3_code"})
+            )
+
+            codes_df = (
+                pd.read_excel(raw_data, sheet_name="Codebook")
+                .rename(
+                    columns={
+                        "variable code": "indicator_code",
+                        "variable name": "indicator_name",
+                    }
+                )
+                .loc[:, ["indicator_code", "indicator_name", "unit", "currency"]]
+                .replace("-", np.nan)
+            )
```

Comment:
Should these two steps happen separately? At least in two different try blocks, but it could also be that you make a data cleaning function (outside the class), with its own error handling, and a codes cleaning function (outside the class), with its own error handling. 
That way if something fails, we would know if it was on reading the data or on cleaning the codebook

Reply:
yes but keep inside the class for visibility to the user